File: PERL/Cap07/c07ex05.pl
# c07ex05.pl

#!/usr/bin/perl

use strict;
use warnings;

sub FATORIAL {
  my ($N, $FAT) = @_;
  for (my $I = 1; $I <= $N; $I++) {
    $$FAT *= $I;
  }
}

my $RESP = 1;

print "Qual fatorial: ";
my $LIMITE = <STDIN>;
chomp $LIMITE;

FATORIAL($LIMITE, \$RESP);

print "Fatorial = $RESP\n";
print "\n";
print "Tecle <Enter> para encerrar... ";
<STDIN>;

File: PASCAL/Cap07/c07ex05.pas
program c07ex05;

procedure FATORIAL(N: Integer; var FAT: Integer);
var
  I: Integer;
begin
  FAT := 1;
  for I := 1 to N do
  begin
    FAT := FAT * I;
  end;
end;

var
  RESP, LIMITE: Integer;

begin
  RESP := 1;
  
  Write('Qual fatorial: ');
  Readln(LIMITE);
  
  FATORIAL(LIMITE, RESP);
  
  Writeln('Fatorial = ', RESP);
  
  Writeln;
  Write('Tecle <Enter> para encerrar... ');
  Readln;
end.
